### DISCUSSION TITLE
Is it possible to use Realm in a Swift package that is referenced by a Swift UI project that does not reference Realm?

### DISCUSSION THREAD

**Discussion by NickOliva** [realm/realm-cocoa#8328] [Q&A]
Hi, all. 

I'm interested in using Realm, but in my architecture the Swift UI project (NickApp) doesn't know how data access occurs. All data calls are made through a referenced Swift package, NickData. 

Suppose my functionality is the same as the sample tasks app on mongo's site. I want NickData to use RealmSync, but not to have any UI. So the Item (or Task) is in NickData and changes to it sync to mongo, etc. NickData maps "Item" to "ItemViewModel" (for example) and gives "ItemViewModel" to NickApp, which displays it in ItemView. 

It seems, however, that Realm Swift is very tightly coupled to Swift UI. Call me old-fashioned, but the idea of my presentation layer knowing the internals of objects that are in my data layer doesn't sit well. Can anyone point me to a working sample solution that is architected as I described? Or, can anyone give me any guidance on what I have to do to make this work?

**Comment by leemaguire** [comment]
You can use Realm in your NickData package and propagate the data to the View. You would need to have the ViewModel be an `ObservableObject` so that when the change listener is called `objectWillChange.send()` can be invoked on each change (this will update the View). You'll also likely need to freeze the Realm objects before passing them to the View (this allows the SwiftUI View diffing functionality to play nicely with Realm.). I'd suggest taking a look at the internals of our property wrappers (`@ObservedResults`, `@ObservedRealmObject`) to get an idea of how we are doing it under the hood.

**Reply by NickOliva** [reply]
Will do, @leemaguire . Thanks.
